### PR TITLE
[CI] only known devs are authorized to trigger CI

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -119,8 +119,18 @@ def go_test_linux() {
 }
 
 def is_authorized(name) {
-  def authorized_user = ['VoVAllen', 'BarclayII', 'jermainewang', 'zheng-da', 'mufeili', 'Rhett-Ying', 'isratnisa']
-  return (name in authorized_user)
+  def devs = ['dgl-bot', 'noreply', 'Rhett-Ying', 'BarclayII', 'jermainewang',
```
Comment (jermainewang) [inline Jenkinsfile:122]: noreply?

Comment (Rhett-Ying) [reply Jenkinsfile:122]: if a PR is merged into master, CI will be triggered. the author is `noreply`...

Comment (Rhett-Ying) [reply Jenkinsfile:122]: I was wrong. In this case, `env.CHANGE_AUTHOR` is null. If it's `null`, let's regard it authenticated?

Comment (jermainewang) [reply Jenkinsfile:122]: Sure. I'm not familiar. Just let you know that master branch is protected, i.e., only authorized users are allowed to merge to it.

File: Jenkinsfile [hunk 2]
```diff
@@ -129,7 +139,52 @@ pipeline {
         issueCommentTrigger('@dgl-bot .*')
   }
   stages {
-    stage('Regression Test Trigger') {
+    stage('Authentication') {
```
Comment (jermainewang) [inline Jenkinsfile:142]: Can we add some comments about what is the logic here?

**Review by jermainewang — APPROVED** [review]

**Pull request merged**